File: internal/dto/lease.go
package dto

type LeaseRef struct {
	Namespace string `json:"namespace"`
	Name      string `json:"name"`
}

type Lease struct {
	Name                 string
	Namespace            string
	HolderIdentity       string
	LeaseDurationSeconds int32
	RenewTime            string
	AcquireTime          string
	LeaseTransitions     int32
	Age                  string
	CreatedAt            string
	Labels               map[string]string
	Annotations          map[string]string
	ManagedFields        []ManagedField
}

File: packages/core/frontend/src/types/resources/lease.ts
import type { ManagedField } from "./shared";

export interface Lease {
  Name: string;
  Namespace: string;
  HolderIdentity: string;
  LeaseDurationSeconds: number;
  RenewTime: string;
  AcquireTime: string;
  LeaseTransitions: number;
  Age: string;
  CreatedAt: string;
  Labels: Record<string, string>;
  Annotations: Record<string, string>;
  ManagedFields: ManagedField[];
}
